src/ai/ai.cpp

```diff
@@ -147,14 +170,27 @@ bool ai_will_accept_alliance(sys::state& state, dcon::nation_id target, dcon::na
 	if(state.world.nation_get_ai_rival(target) && state.world.nation_get_ai_rival(target) == state.world.nation_get_ai_rival(from))
 		return true;
 	
+	// Having a mutual, non-rival enemy - incentivizing AI to clump against a
+	// common enemy more easily
+	bool mutual_enemy = false;
+	state.world.for_each_nation([&](dcon::nation_id n) {
+		if(n != target && n != from) {
+			bool enemy_a = military::can_use_cb_against(state, from, n) || military::are_at_war(state, from, n);
```

Comment:
can_use_cb_against is *expensive*. And this is an n^2 check. So you must remove this